db.go

```diff
@@ -131,6 +136,34 @@ func (rr *redisRepo) fetchInstanceEvents(instanceID string) ([]*lifecycleEvent,
 	return events, nil
 }
 
+func (rr *redisRepo) fetchAllInstanceEvents() (map[string][]*lifecycleEvent, error) {
+	conn := rr.cg.Get()
+	defer rr.closeConn(conn)
+
+	instanceEventKeys, err := redis.Strings(conn.Do("KEYS", fmt.Sprintf("%s:instance:*:events", RedisNamespace)))
```

Comment:
I'm generally not a fan of KEYS because with the single-threaded design of redis it means that even a single slow command can completely kill the server.

I think SCAN is an acceptable workaround in this case.